File: FastMines_core/src/fmg/core/app/serializers/StatisticsSerializer.java
package fmg.swing.app.serializers;

import java.io.IOException;
import java.io.ObjectInput;
import java.io.ObjectOutput;

import fmg.core.app.ISerializer;
import fmg.core.types.model.Statistics;

public class StatisticsSerializer implements ISerializer {

    private static final long VERSION = 1;

    public void write(Statistics data, ObjectOutput to) throws IOException {
        to.writeLong(VERSION);
        to.writeLong(data.gameNumber);
        to.writeLong(data.gameWin);
        to.writeLong(data.openField);
        to.writeLong(data.playTime);
        to.writeLong(data.clickCount);
    }

    public void read(Statistics to, ObjectInput from) throws IOException {
        long version = from.readLong();
        if (version != VERSION)
            throw new RuntimeException("Unsupported " + Statistics.class.getSimpleName() + " version " + version);

        to.gameNumber = from.readLong();
        to.gameWin    = from.readLong();
        to.openField  = from.readLong();
        to.playTime   = from.readLong();
        to.clickCount = from.readLong();
    }

}

File: FastMines_core/src/fmg/core/app/serializers/StatisticsSerializer.cs
using System;
using System.IO;
using Fmg.Core.App.Model;

namespace Fmg.Core.App.Serializers {

    /// <summary> Statistics (de)serializer </summary>
    public class StatisticsSerializer : ISerializer {

        private const long VERSION = 1;

        public void Write(Statistics data, BinaryWriter to) {
            to.Write(VERSION);
            to.Write(data.gameNumber);
            to.Write(data.gameWin);
            to.Write(data.openField);
            to.Write(data.playTime);
            to.Write(data.clickCount);
        }

        public void Read(Statistics to, BinaryReader from) {
            long version = from.ReadInt64();
            if (version != VERSION)
                throw new ArgumentException("Unsupported " + nameof(Statistics) + " version " + version);

            to.gameNumber = from.ReadInt64();
            to.gameWin    = from.ReadInt64();
            to.openField  = from.ReadInt64();
            to.playTime   = from.ReadInt64();
            to.clickCount = from.ReadInt64();
        }

}

}
